x/concentrated-liquidity/internal/math/math.go

```diff
@@ -81,20 +95,21 @@ func CalcAmount1Delta(liq, sqrtPriceA, sqrtPriceB sdk.Dec, roundUp bool) sdk.Dec
 // else
 // sqrtPriceNext = ((liquidity)) / (((liquidity) / (sqrtPriceCurrent)) + (amountRemaining))
 func GetNextSqrtPriceFromAmount0RoundingUp(sqrtPriceCurrent, liquidity, amountRemaining sdk.Dec) (sqrtPriceNext sdk.Dec) {
-	numerator := liquidity
-	product := amountRemaining.Mul(sqrtPriceCurrent)
+	if amountRemaining.Equal(sdk.ZeroDec()) {
+		return sqrtPriceCurrent
+	}
 
-	if product.Quo(amountRemaining).Equal(sqrtPriceCurrent) {
-		denominator := numerator.Add(product)
-		if denominator.GTE(numerator) {
-			numerator = numerator.Mul(sqrtPriceCurrent)
-			sqrtPriceNext = numerator.QuoRoundUp(denominator)
-			return sqrtPriceNext
+	product := amountRemaining.Mul(sqrtPriceCurrent)
+	// use precise formula if product doesn't overflow
+	// TODO: see if we even need to do this check
```

Comment:
Wdyt about creating an issue for this regarding adding testcase for testing product overflow?

Reply:
https://github.com/osmosis-labs/osmosis/issues/3791